sql/src/main/java/org/apache/druid/sql/calcite/aggregation/builtin/StringSqlAggregator.java

```diff
@@ -0,0 +1,219 @@
+      RexBuilder rexBuilder,
+      String name,
+      AggregateCall aggregateCall,
+      Project project,
+      List<Aggregation> existingAggregations,
+      boolean finalizeAggregations
+  )
+  {
+    final List<DruidExpression> arguments = aggregateCall
+        .getArgList()
+        .stream()
+        .map(i -> Expressions.fromFieldAccess(rowSignature, project, i))
+        .map(rexNode -> Expressions.toDruidExpression(plannerContext, rowSignature, rexNode))
+        .collect(Collectors.toList());
+
+    if (arguments.stream().anyMatch(Objects::isNull)) {
+      return null;
+    }
+
+    RexNode separatorNode = Expressions.fromFieldAccess(
+        rowSignature,
+        project,
+        aggregateCall.getArgList().get(1)
+    );
+    if (!separatorNode.isA(SqlKind.LITERAL)) {
+      // separator must be a literal
+      return null;
+    }
+    String separator = RexLiteral.stringValue(separatorNode);
+
+    if (separator == null) {
+      // separator must not be null
+      return null;
+    }
+
+    Integer maxSizeBytes = null;
+    if (arguments.size() > 2) {
+      RexNode maxBytes = Expressions.fromFieldAccess(
+          rowSignature,
+          project,
+          aggregateCall.getArgList().get(2)
+      );
+      if (!maxBytes.isA(SqlKind.LITERAL)) {
+        // maxBytes must be a literal
+        return null;
+      }
+      maxSizeBytes = ((Number) RexLiteral.value(maxBytes)).intValue();
+    }
+    final DruidExpression arg = arguments.get(0);
+    final ExprMacroTable macroTable = plannerContext.getExprMacroTable();
+
+    final String initialvalue = "[]";
+    final ValueType elementType = ValueType.STRING;
+    final String fieldName;
+    if (arg.isDirectColumnAccess()) {
+      fieldName = arg.getDirectColumn();
+    } else {
+      VirtualColumn vc = virtualColumnRegistry.getOrCreateVirtualColumnForExpression(plannerContext, arg, elementType);
+      fieldName = vc.getOutputName();
+    }
+
+    final String finalizer = StringUtils.format("if(array_length(o) == 0, null, array_to_string(o, '%s'))", separator);
+    final NotDimFilter dimFilter = new NotDimFilter(new SelectorDimFilter(fieldName, null, null));
+    if (aggregateCall.isDistinct()) {
+      return Aggregation.create(
+          // string_agg ignores nulls
+          new FilteredAggregatorFactory(
+              new ExpressionLambdaAggregatorFactory(
+                  name,
+                  ImmutableSet.of(fieldName),
+                  null,
+                  initialvalue,
+                  null,
+                  StringUtils.format("array_set_add(\"__acc\", \"%s\")", fieldName),
+                  StringUtils.format("array_set_add_all(\"__acc\", \"%s\")", name),
+                  null,
+                  finalizer,
+                  maxSizeBytes != null ? new HumanReadableBytes(maxSizeBytes) : null,
+                  macroTable
+              ),
+              dimFilter
+          )
+      );
+    } else {
+      return Aggregation.create(
```

Comment:
the if-else can be collapsed

Reply:
yeah, it could if i pull out the two expressions into a separate conditional block; I did not make this change yet because it seems like minor shifting of the condition but will modify here (as well as [ARRAY_AGG](https://github.com/apache/druid/blob/master/sql/src/main/java/org/apache/druid/sql/calcite/aggregation/builtin/ArraySqlAggregator.java#L143) since I did the same thing there) if you feel more strongly about it.